docs/rfc/008-built-in-declarations.md

```diff
@@ -41,42 +36,33 @@ Leo supports four kinds of top-level declarations:
 
 Leaving import declarations aside for the moment since they are "meta" in some sense
 (as they bring in names of entities declared elsewhere),
-it may make sense for any of the four kinds of declarations above to have built-in instances,
-i.e. we could have some built-in functions, circuit types, global constants, and type aliases.
-This is why this RFC talks of built-in declarations, more broadly than just built-in type aliases that inspired it.
+it may make sense for any of the four kinds of declarations above to have built-in instances, i.e., we could have some built-in functions, circuit types, global constants, and type aliases. These features are why this RFC talks of built-in declarations, more broadly than just built-in type aliases that inspired it.
 
-The built-in status of the envisioned declarations could be achieved in slightly different ways:
-1. Their names could be simply available in any program,
-   without any explicit declaration found anywhere for them.
-2. They could be declared in some core library files explicitly,
-   and be available in any program without needing to be explicitly import them,
-   like `java.lang.String` in Java or `std::Option` in Rust.
-3. They could be declared in some core library files explicitly,
-   and be available only in programs that explicitly import them.
+The built-in status of the envisioned declarations will be done through explicitly declared core library files. Then these core library files must be explicitly imported. This way helps avoid unnecessary code bloat in the compilation, and any user asked for AST snapshots.
 
-From a user's perspective, there is not a lot of difference between cases 1 and 2 above:
-in both cases, the names are available; the only difference is that in case 2 the user can see the declaration somewhere.
+## Drawbacks
+
+This does not seem to bring any drawbacks.
 
-Also note that case 2 could be seen as having an implicit (i.e. built-in) import of the library/libraries in question.
-Again, imports are "meta" in this context, and what counts are really the other kinds of declarations.
+## Effect on Ecosystem
 
-In cases 2 and 3, a related but somewhat independent issue is whether those declarations have Leo definitions or not.
-The Leo library already includes functions like the one for BLAKE2s that are not defined in Leo,
-but rather "natively" in Rust/R1CS.
+This change may interact with libraries and packages in some way,
+if we go with case 2 or 3 above.
```

Comment:
The list of cases 1, 2, 3 was removed, so this should be rephrased.

Reply:
The same applies to other references to cases 1/2/3 below.

Reply:
Ah whoops I missed some of these.

Reply:
Resolved.